pgjdbc/src/main/java/org/postgresql/core/AuthenticationPluginManager.java

```diff
@@ -67,14 +71,14 @@ private AuthenticationPluginManager() {
    * @throws PSQLException Throws a PSQLException if the plugin class cannot be instantiated or if the retrieved password is null.
    */
   public static byte[] getEncodedPassword(AuthenticationRequestType type, Properties info) throws PSQLException {
```

Comment:
Does it really belong to the public API?
Could we keep a single `char @Nullable [] getPassword` and be done with it?

`getEncodedPassword` belongs to internal helpers rather than to the public API.

Reply:
The manager class isn't part of the public API. Only the AuthenticationPlugin interface which only has the `char[]` method.